remotes/handlers.go

```diff
@@ -56,11 +56,22 @@ func FetchHandler(ingester content.Ingester, fetcher Fetcher) images.HandlerFunc
 func fetch(ctx context.Context, ingester content.Ingester, fetcher Fetcher, desc ocispec.Descriptor) error {
 	log.G(ctx).Debug("fetch")
 	ref := MakeRefKey(ctx, desc)
+
+	cw, err := ingester.Writer(ctx, ref, desc.Size, desc.Digest)
```

Comment:
I'm hitting an issue with concurrent image pulling. If pulling the same resource concurrently in different operations, only one `Dispatch` could proceed, and others will return error.

The reason is that `Writer` never returns `ErrExists` currently.

Should we let it return `ErrExists` when it's already locked? https://github.com/containerd/containerd/blob/master/content/store.go#L238